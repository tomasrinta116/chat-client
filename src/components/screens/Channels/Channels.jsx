import React from "react";
import { withStyles } from "@material-ui/core/styles";
import ChannelList from "./ChannelList/ChannelList";
import CreateChannelButton from "./CreateChannelButton/CreateChannelButton";
import NavBar from "../../common/NavBar/NavBar";
import styles from "./Channels.styles";
import PropTypes from 'prop-types';

class Channels extends React.Component {
  state = {
    channels: []
  };

  async componentDidMount() {
    const { sb } = this.props;
    const openChannelListQuery = sb.OpenChannel.createOpenChannelListQuery();
    openChannelListQuery.next((channels, error) => {
      if (error) return console.log(error);
      this.setState({ channels: channels });
    });
  }

  enterChannel = channelURL => {
    const { sb } = this.props;
    return new Promise(async resolve => {
      sb.OpenChannel.getChannel(channelURL, (channel, error) => {
        if (error) return console.log(error);
        channel.enter((response, error) => {
          if (error) return console.log(error);
          console.log(channel);
          resolve(channel);
        });
      });
    });
  };

  render() {
    const {
      classes: { channelsContainer, createChannel, channelList },
      history,
      sb
    } = this.props;
    const { channels } = this.state;
    return (
      <React.Fragment>
        <NavBar history={history} sb={sb}/>
        <div className={channelsContainer}>
          <div className={channelList}>
            <ChannelList
              history={history}
              channels={channels}
              enterChannel={this.enterChannel}
            />
          </div>
          <div className={createChannel}>
            <CreateChannelButton
              sb={sb}
              history={history}
              enterChannel={this.enterChannel}
            />
          </div>
        </div>
      </React.Fragment>
    );
  }
}

Channels.propTypes = {
  sb: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
}

export default withStyles(styles)(Channels);
