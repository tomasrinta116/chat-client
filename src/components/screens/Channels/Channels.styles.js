const styles = {
  /* --- <Channels /> --- */

  channelsContainer: {
    width: "100%",
    display: "flex",
    direction: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  channelList: {
    width: "75%",
    minWidth: 300,
    maxWidth: 600,
    padding: "3rem",
    justifyContent: "center"
  },
  createChannel: {
    position: "absolute",
    bottom: 0,
    right: 0,
    margin: 15
  },
};

export default styles;