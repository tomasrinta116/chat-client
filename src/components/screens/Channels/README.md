# Channels Overview

This folder contains all of the files and components for the Channels screen. This is the landing page for a user once they login, and presents them with the ability to select a pre-existing channel and redirect to that channel's chat interface, or create a new channel. 

All channels in the current version of this app are open. This means that anyone can access them and chat. However, only a creator of a channel or an admin can ultimately delete a channel. 