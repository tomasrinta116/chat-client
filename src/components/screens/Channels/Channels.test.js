import React from 'react';
import {shallow} from 'enzyme';
import Channels from './Channels';

describe("<Channels />", () => {
    let wrapper;
    const props = {
        sb: {},
        history: {},
    }
    beforeEach(() => {
        wrapper = shallow(<Channels {...props} />).dive();
    })

    test("does it render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
})