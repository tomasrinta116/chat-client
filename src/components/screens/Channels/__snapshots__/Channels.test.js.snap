// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Channels /> does it render correctly 1`] = `
<Fragment>
  <WithStyles(Navbar)
    history={Object {}}
    sb={Object {}}
  />
  <div
    className="Channels-channelsContainer-1"
  >
    <div
      className="Channels-channelList-2"
    >
      <WithStyles(ChannelList)
        channels={Array []}
        enterChannel={[Function]}
        history={Object {}}
      />
    </div>
    <div
      className="Channels-createChannel-3"
    >
      <CreateChannelButton
        enterChannel={[Function]}
        history={Object {}}
        sb={Object {}}
      />
    </div>
  </div>
</Fragment>
`;
